bdk_core/src/sparse_chain.rs

```diff
@@ -629,107 +672,67 @@ impl<E: core::fmt::Debug> Display for MergeFailure<E> {
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct ChainIndex<E = ()> {
-    pub height: TxHeight,
-    pub extension: E,
+    /// Obtain the index's lower bound of a given height.
+    fn min_ord_of_height(height: TxHeight) -> Self;
 }
 
-impl From<TxHeight> for ChainIndex {
-    fn from(height: TxHeight) -> Self {
-        Self {
-            height,
-            extension: (),
-        }
-    }
-}
-
-impl<E: ChainIndexExtension> From<(TxHeight, E)> for ChainIndex<E> {
-    fn from((height, extension): (TxHeight, E)) -> Self {
-        Self { height, extension }
-    }
+fn min_txid() -> Txid {
+    Txid::from_inner([0x00; 32])
 }
 
-/// Represents the height in which a transaction is confirmed at.
-#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub enum TxHeight {
-    Confirmed(u32),
-    Unconfirmed,
+fn max_txid() -> Txid {
+    Txid::from_inner([0xff; 32])
 }
 
-impl Display for TxHeight {
-    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
-        match self {
-            Self::Confirmed(h) => core::write!(f, "confirmed_at({})", h),
-            Self::Unconfirmed => core::write!(f, "unconfirmed"),
-        }
-    }
-}
+#[cfg(test)]
+pub mod verify_chain_index {
+    use alloc::vec::Vec;
 
-impl From<Option<u32>> for TxHeight {
-    fn from(opt: Option<u32>) -> Self {
-        match opt {
-            Some(h) => Self::Confirmed(h),
-            None => Self::Unconfirmed,
-        }
-    }
-}
+    use super::ChainIndex;
+    use crate::TxHeight;
 
-impl From<TxHeight> for Option<u32> {
-    fn from(height: TxHeight) -> Self {
-        match height {
-            TxHeight::Confirmed(h) => Some(h),
-            TxHeight::Unconfirmed => None,
+    pub fn verify_chain_index<I: ChainIndex>(head_count: u32, tail_count: u32) {
+        let values = (0..head_count)
+            .chain(u32::MAX - tail_count..u32::MAX)
+            .flat_map(|i| {
+                [
+                    I::min_ord_of_height(TxHeight::Confirmed(i)),
+                    I::max_ord_of_height(TxHeight::Confirmed(i)),
+                ]
+            })
+            .chain([
+                I::min_ord_of_height(TxHeight::Unconfirmed),
+                I::max_ord_of_height(TxHeight::Unconfirmed),
+            ])
+            .collect::<Vec<_>>();
+
+        for i in 0..values.len() {
+            for j in 0..values.len() {
+                if i == j {
+                    assert_eq!(values[i], values[j]);
+                }
+                if i < j {
+                    assert!(values[i] <= values[j]);
+                }
+                if i > j {
+                    assert!(values[i] >= values[j]);
+                }
+            }
         }
     }
-}
 
-impl TxHeight {
-    pub fn is_confirmed(&self) -> bool {
-        matches!(self, Self::Confirmed(_))
+    #[test]
+    fn verify_tx_height() {
+        verify_chain_index::<TxHeight>(1000, 1000);
```

Comment:
I guess do this for confirmation time too? I think it would be better to put this test in `chain_data.rs` because that's where the impls are.